apps/hs/helen/src/Helen/Core.hs

```diff
@@ -79,9 +89,38 @@ initHelen = do
         STM.takeTMVar clientStore >>=
           STM.putTMVar clientStore . HashMap.update (\(c, u) -> Just (c, u >> action cId)) cId
     , _serviceManager = defServiceManager
-    , trustedClients = Set.empty
+    , trustedClients = confTrustedClients conf
     }
 
+readConfigFromYamlFile :: IO Config
+readConfigFromYamlFile = do
+  cf <- resolveFile' "helen-config.yaml"
+  errOrConfig <- readYamlSafe cf
+  case errOrConfig of
+    Left err -> do
+      putStrLn $ unlines
```

Comment:
Do you mean `HelenWorld` cannot be initialized before you run `readConfigFromYamlFile`, and that is the reason you don't use MonadLogger here?